blog/tech-talk-window-resize-behavior.md

```diff
@@ -0,0 +1,290 @@
+---
+title: 'Tech Talk #1: Improving Window Resize Behavior'
+date: 2025-12-19T00:00:00.000Z
+authors:
+  - name: nikwen
+    url: 'https://github.com/nikwen'
+    image_url: 'https://github.com/nikwen.png?size=96'
+slug: tech-talk-window-resize-behavior
+tags: [techtalk, internals]
+---
+
+_We're launching a new blog post series where we share glimpses into our work on Electron. If you find this work interesting, please consider [contributing](https://github.com/electron/electron/)!_
+
+---
+
+Recently, I worked on improving Electron and Chromium's window resize behavior.
+
+## The bug
+
+We were seeing an issue on Windows where old frames would become visible while resizing a window:
+
+![Animated GIF showing the issue where old frames would be shown while resizing windows](/assets/img/blog/tech-talk-window-resize-behavior/window-resize-bug-showing-old-frames.gif)
+
+## What made this bug particularly interesting?
+
+1. It was challenging.
+2. As you'll see later, there were two different bugs under the hood.
+
+## Fixing the bug
+
+With a bug like this, the first challenge is figuring out where to start looking.
+
+Electron builds upon Chromium. Chromium is the open source version of Google Chrome. When compiling Electron, Electron's source code is added into the Chromium source tree as a subdirectory. Electron then relies on Chromium's code to provide most of the functionality of a modern browser.
+
+Chromium has about 36 million lines of code. Electron is a large project, too. That is a lot of code that could be causing this issue.
+
+<!--truncate-->
+
+## Narrowing down the root cause
+
+I did a lot of experimentation.
+
+First, I noticed that the issue occurred in Google Chrome, too:
+
+![Screenshot of Google Chrome also showing the resize issue](/assets/img/blog/tech-talk-window-resize-behavior/google-chrome-repro.png)
+
+This suggested that the issue was likely in Chromium, not in Electron.
+
+Additionally, the issue was not visible on macOS. That suggested that it was in Windows-specific source code.
+
+## The crucial lead
+
+I tried a lot of different command line flags and configuration options.
+
+I noticed that [`app.disableHardwareAcceleration()`](../docs/latest/api/app#appdisablehardwareacceleration) fixed the issue. Without hardware acceleration, the issue was gone.
+
+Here is some context: Chromium supports various different graphics APIs for showing pixels on screen (OpenGL, Vulkan, Metal, and more). On Windows, it uses different graphics APIs than on Windows or Linux. Even on Windows, Chromium can work with multiple different graphics backends.
+
+Which graphics backend Chromium uses depends on the user's hardware. For example, some graphics backends require the computer to have a GPU.
+
+I tried various graphics backends and noticed that the following flags fixed the issue:
+
+- `--use-angle=warp`
+- `--use-angle=vulkan`
+- `--use-gl=desktop`
+- `--use-gl=egl`
+- `--use-gl=osmesa`
+- `--use-gl=swiftshader`
```

Comment:
In the narrative, I think you could explain what the difference between calling `--use-angle` and `--use-gl` represents for Chromium and avoid needing to explain ANGLE at the bottom after we've already name-dropped it a few times.

As an aside, TIL that ANGLE stands for "Almost Native Graphics Layer Engine". Funny name!